### DISCUSSION TITLE
STM-PE : To enforce or not to enforce?

### DISCUSSION THREAD

**Discussion by tlaurion** [linuxboot/heads#1747] [General]
- [Firmware Security blog “Stateless Laptop” tag](https://firmwaresecurity.com/tag/stateless-laptop/)
  - (2015) [Joanna Rutoswka - State considered harmful - A proposal for a stateless laptop (new paper + conference) – Invisible Labs blog](https://blog.invisiblethings.org/2015/12/23/state_harmful.html)
  - (2015) [Johanna Rutowska paper: States considered harmful](https://blog.invisiblethings.org/papers/2015/state_harmful.pdf)
  - (2015) [32c3 - lecture: Towards (reasonably) trustworthy x86 laptops](https://fahrplan.events.ccc.de/congress/2015/Fahrplan/events/7352.html) 
    - TXT untrusted, vt-d promises flawed
      - To be “trusted”, would require FOSS SMM to be jailed behind SMI Transfer Mode (STM)
        - (2019) [OSFC conference - NSA - Eugene Myers - Implementing STM support for Coreboot](https://www.osfc.io/2019/talks/implementing-stm-support-for-coreboot/)
        - (2019) [Digestible hig-level article](https://cyberscoop.com/nsa-firmware-open-source-coreboot-stm-pe-eugene-myers/)
        - (2019) [Code merged under coreboot](https://review.coreboot.org/c/blobs/+/34026) 
        - STM-PE still not widely deployed in the wild
- (2017) [Stateless laptop (QubesOS blog post - level 2 certification envisioned for 2019)](https://www.qubes-os.org/news/2017/07/08/toward-a-reasonably-secure-laptop/)
  - Never happened. State still considered harmful and untrustworthy.

- (2019) [Trammel Hudson - SpiSpy - Open Source SPI emulation](https://trmm.net/Spispy/)
  - (2019) [Trammel Hudson - TOCTU blog post (and linked lecture)](https://trmm.net/TOCTOU/)
    - Time of Check Time of Use (TOCTU)
      - Lots of vulnerability discovered, showing Bootguard failing promises.
- (2019) [Trammel Hudson & Peter Bosch - HITB2019AMS – Lecture: TOCTOU Attacks Against Secure Boot And BootGuard](https://www.youtube.com/watch?v=hx9MS1_1e2c)
  - (2019) [Peter Bosch - Boot Guard TOCTOU CVE-2019-11098 – blog post](https://pbx.sh/efitoctou/)
  - (2019) [Trammel Hudson – SpiSpy - Lecture: SPI flash device emulation Open source tools for flash emulation and research](https://media.ccc.de/v/Camp2019-10186-spispy_spi_flash_device_emulation)
    - Permits to see what happens from PCH to SPI
      - TOCTU proofs given. Speakers made to beep while bootguard was on (tampering of bootguard proven)
        - So.... What do we do with failed premises? 

----

So what do we do now?
  - To have VT-D/TXT/DRTM promises, STM-PE is needed.
    - Do we trust STM-PE

**Comment by tlaurion** [comment]
@marmarek ? ping you to here off channel

**Comment by arhabd** [comment]
(2018) [STM/PE & XHIM — Eugene Myers, NSA — Platform Security Summit 2018](https://invidious.darkness.services/watch?v=VMBOLuN1gdU)
(2018) [Using the Intel STM for Protected Execution](https://www.platformsecuritysummit.com/2018/speaker/myers/STMPE2Intelv84a.pdf)
(2022) [Dasharo issue #82 Coreboot, SMM (System Management Mode) and performance differences](https://github.com/Dasharo/dasharo-issues/issues/82)
(2018) [SMM-based Runtime Integrity Measurement — Brian Delgado, Intel — Platform Security Summit 2018](https://invidious.darkness.services/watch?v=DIId8T_jSUo)
(??)[STM-PE on github](https://github.com/EugeneDMyers/STM/tree/stmpe)
(2022) [Dasharo issue #132 Intel STM/PPAM support](https://github.com/Dasharo/dasharo-issues/issues/132)

**Reply by arhabd** [reply]
(2013) [Performance Implications of System Management Mode](https://web.cecs.pdx.edu/~karavan/research/SMM_IISWC_preprint.pdf)

**Reply by arhabd** [reply]
(2011) [SICE: a hardware-level strongly isolated computing environment for x86 multi-core platforms](https://static.aminer.org/pdf/20170130/pdfs/ccs/vysnbuf9rtk5qx2lj3awownesdmvh7e6.pdf)

**Reply by arhabd** [reply]
(2023) [Ringhopper - How We Almost Zero day'd the World](https://www.youtube.com/watch?v=u8V4ofWpHZk)

**Reply by arhabd** [reply]
(2022) [Finding SMM Privilege-Escalation Vulnerabilities in
UEFI Firmware with Protocol-Centric Static Analysis](https://ieeexplore.ieee.org/stampPDF/getPDF.jsp?arnumber=9833723)

**Reply by tlaurion** [reply]
(2025+) https://github.com/9elements/LinuxBootSMM

**Comment by tlaurion** [comment]
2024 On MSI setup guide https://github.com/Dasharo/dasharo-issues/issues/132#issuecomment-2083672435

Traces as well for OptiPlex 9010 https://github.com/Dasharo/dasharo-issues/issues/132#issuecomment-2258451486


Direct input for https://github.com/linuxboot/heads/issues/1739

**Comment by tlaurion** [comment]
@marmarek ping again
